### ISSUE TITLE
Add a "submit project" option

### ISSUE THREAD

**Issue by javierguerragiraldez** [nektro/aquila#16]
All the login options are Git hosting.  I created an account at codeberg.org just to be able to login, as I don't intend to host any project there.  But the dashboard is empty, Is it just listing my codeberg account?  My projects are hosted and published elsewhere, but I make them accessible both via Fossil and by zipfile downloads.  Where can I submit an URL to the zipfile?

**Comment by nektro** [comment]
yes the dashboard is only showing your actively imported projects from [provider]. what site do you use fossil hosting with and do they support OAuth2 login?

**Comment by javierguerragiraldez** [comment]
I use chisel (https://chiselapp.com/user/javier/), but I don't think it supports OAuth.  Besides, it's just a slight convenience to use it, since Fossil makes it absolutely trivial to host on your own.   That's why I asked for a way to submit an entry directly, and not have it tied to the login. 

Also because it could be convenient to publish the zipfile URL, not just the main fossil repo.  Both because Fossil support isn't yet merged in zigmod, and because it would be nice to allow people to download a module without having to first install Fossil.

Finally, it's not inconceivable that I'd prefer to use codeberg for login but still want to submit a github project (If i hadn't already migrated all my personal projects off).

**Comment by nektro** [comment]
> That's why I asked for a way to ... not have it tied to the login.

this was avoided intentionally in the site design to assert ownership of packages

> Also because it could be convenient to publish the zipfile URL

this was avoided intentionally in the site design for security reasons, package revisions must be able to be backed by the origin

> it would be nice to allow people to download a module without having to first install Fossil.

in a scenario where it would be possible to add chisel login support, they would. Zigmod still always fetches from upstream using the original download method

**Comment by nektro** [comment]
not to say that the first two would never be possible, just not on this app in particular. i'm still open to the community's discussion on a new site. only time will tell how aquila moves forward longterm

**Comment by javierguerragiraldez** [comment]
I understand that ownership assertion is very complicated; the idea of using the login from the repo hosting is quite ingenious and probably covers the vast majority of cases.
Unfortunately, delegating it to a handful of centralized services means that truly decentralized code management gets excluded.

Perhaps the easiest way to not depend on the login provider would be to make aquila generate an HMAC or maybe just a random key, then the user (me, in this case) would include it in the `zig.mod` file.
something like this:
- I login (using any OAuth2 provider)
- the project I want to submit doesn't appear in the Dashboard (because I don't host it with the login provider)
- I press a "submit project hosted elsewhere" button
- Aquila presents a form that shows a uniquely generated key and a field for me to enter a URL
- I add a `key: xxxxx` field to the `zig.mod` file in my project and commit/push the change.
- I enter the URL to the project (or just the zipfile, if zigmod doesn't support the SCM I'm using)
- Aquila verifies the key in the project matches the one it generated, accepting the project if positive.

would that (or something similar) be OK with you?
